src/test/java/uk/gov/justice/probation/courtcaseservice/service/OffenderUpdateServiceTest.java

```diff
@@ -39,23 +40,21 @@ void shouldReturnEmptyOffenderWhenDefendantCrnIsAbsent() {
         given(defendantRepository.findFirstByDefendantIdOrderByIdDesc(DEFENDANT_ID))
                 .willReturn(Optional.of(DefendantEntity.builder().build()));
 
-        var offender = offenderUpdateService.getDefendantOffenderByDefendantId(DEFENDANT_ID).block();
-        assertThat(offender).isEqualTo(OffenderEntity.builder().build());
-        verify(defendantRepository).findFirstByDefendantIdOrderByIdDesc(DEFENDANT_ID);
-        verifyNoInteractions(offenderRepository);
+        assertThatExceptionOfType(EntityNotFoundException.class)
```

Comment:
Just a general comment about this class - we've now got the concept of an Offender both as a domain Entity and as data we fetch and aggregate from Community Api. I wonder whether at some point we'd benefit from making a clearer distinction between the two 🤔  One to ponder as a future improvement maybe!